runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -254,7 +254,8 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 	{
 		final String menuOption = event.getMenuOption();
 
-		if (CLEAR_HISTORY.equals(menuOption))
+		// The menu option for clear history is "<col=ffff00>Public:</col> Clear history"
+		if (menuOption.contains(CLEAR_HISTORY))
```

Comment:
This needs to check better if the right menu option is clicked on the right widgets. As is, any menu options containing `Clear history` will clear the chat history.